### DISCUSSION TITLE
TCP/IP Socket connection

### DISCUSSION THREAD

**Discussion by D49ll** [PlusToolkit/PlusLib#1113] [General]
Hi,

I'm currently in the process of creating a new optical tracking system, i.e. I'm not using established systems like NDI's Polaris. 

However, I have access to my hardware's SDK which sends positional and rotation as a homogeneous matrix with the help of tcp/ip-socket communication.

I've tried to find information about how to add new input devices, but have not found a direct TCP/IP page.

Do anyone have any recommendations as to where i should begin towards adding a new device?

-D

**Comment by adamrankin** [comment]
This page will help you with the coding infrastructure towards adding a new device:
https://plustoolkit.github.io/devicecode

The NDICAPI uses a network connection to their newer devices so you could use that device as an example.
